pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator/assets"
+)
+
+// webhookAuthenticatorController makes sure that the webhook token authenticators
+// are configured so that they point to the oauth-apiserver's tokenvalidation
+// endpoint. That is, if the type of authentication in the auth config is set t
+// be the integrated authentication
+type webhookAuthenticatorController struct {
+	authentication  configv1client.AuthenticationInterface
+	serviceAccounts corev1client.ServiceAccountsGetter
+
+	saLister  corev1listers.ServiceAccountLister
+	svcLister corev1listers.ServiceLister
+	secrets   corev1client.SecretsGetter
+
+	operatorConfigClient operatorconfigclient.AuthenticationsGetter
+	operatorClient       v1helpers.OperatorClient
+}
+
+func NewWebhookAuthenticatorController(
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	configInformer configinformers.SharedInformerFactory,
+	secrets corev1client.SecretsGetter,
+	serviceAccounts corev1client.ServiceAccountsGetter,
+	authentication configv1client.AuthenticationInterface,
+	operatorConfigClient operatorconfigclient.AuthenticationsGetter,
+	operatorClient v1helpers.OperatorClient,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &webhookAuthenticatorController{
+		secrets:              secrets,
+		serviceAccounts:      serviceAccounts,
+		svcLister:            kubeInformersForTargetNamespace.Core().V1().Services().Lister(),
+		saLister:             kubeInformersForTargetNamespace.Core().V1().ServiceAccounts().Lister(),
+		authentication:       authentication,
+		operatorConfigClient: operatorConfigClient,
+		operatorClient:       operatorClient,
+	}
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().ServiceAccounts().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Services().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Secrets().Informer(),
+		configInformer.Config().V1().Authentications().Informer(),
+	).ResyncEvery(30*time.Second).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ToController("WebhookAuthenticatorController", recorder.WithComponentSuffix("webhook-authenticator-controller"))
+}
+
+func (c *webhookAuthenticatorController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	authConfig, err := c.authentication.Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if authConfig.Spec.Type != configv1.AuthenticationTypeIntegratedOAuth && len(authConfig.Spec.Type) != 0 {
+		// not using integrated openshift auth
+		return nil
+	}
+
+	oauthAPIsvc, err := c.svcLister.Services("openshift-oauth-apiserver").Get("api")
```

Comment:
no need to check for this.  Just assume it's always there.  It's created in this operator and the worst you'll do is fail a token authentication that would already have failed.

Reply:
this is there only to get the cluster IP of the service since service hostnames don't seem to be resolvable from KAS

Reply:
> this is there only to get the cluster IP of the service since service hostnames don't seem to be resolvable from KAS

ok.  The cert is probably only signed for dns names.  You're overriding servername?